framework/test/unit/modules/random/stores/used_hash_onion_store.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { StateStore } from '@liskhq/lisk-chain';
+import { InMemoryDatabase } from '@liskhq/lisk-db';
+import {
+	UsedHashOnionsStore,
+	UsedHashOnionStoreObject,
+} from '../../../../../src/modules/random/stores/used_hash_onions';
+
+describe('UsedHashOnionStore', () => {
+	let db: InMemoryDatabase;
+	let offchainStore: StateStore;
+	let usedHashOnionStore: UsedHashOnionsStore;
+	let context: { getOffchainStore: (storePrefix: Buffer, subStorePrefix: Buffer) => StateStore };
+
+	const address = Buffer.from('10');
+
+	beforeEach(async () => {
+		db = new InMemoryDatabase();
+		offchainStore = new StateStore(db);
+		usedHashOnionStore = new UsedHashOnionsStore('random');
+		context = {
+			getOffchainStore: (storePrefix: Buffer, subStorePrefix: Buffer) =>
+				offchainStore.getStore(storePrefix, subStorePrefix),
+		};
+	});
+
+	describe('filterUsedHashOnion', () => {
+		it('if highest usedHashOnion has height less than finalizedHeight it is included in the result, rest are removed', async () => {
+			const defaultUsedHashOnionStoreObject: UsedHashOnionStoreObject = {
+				usedHashOnions: [
+					{
+						count: 5,
+						height: 9,
+					},
+					{
+						count: 6,
+						height: 8,
+					},
+				],
+			};
+
+			const expectedUsedHashOnionStoreObject: UsedHashOnionStoreObject = {
+				usedHashOnions: [
+					{
+						count: 5,
+						height: 9,
+					},
+				],
+			};
+
+			const actualUsedHashOnionStoreObject = usedHashOnionStore.filterUsedHashOnions(
+				defaultUsedHashOnionStoreObject.usedHashOnions,
+				10,
+			);
+
+			expect(actualUsedHashOnionStoreObject).toEqual(expectedUsedHashOnionStoreObject);
+		});
+	});
+
+	describe('setLatest', () => {
+		const defaultUsedHashOnionStoreObject: UsedHashOnionStoreObject = {
+			usedHashOnions: [
+				{
+					count: 5,
+					height: 9,
+				},
+				{
+					count: 6,
+					height: 10,
+				},
+			],
+		};
+
+		const upsertedFilteredUsedHashOnionStoreObject: UsedHashOnionStoreObject = {
+			usedHashOnions: [
+				{
+					count: 6,
+					height: 11,
+				},
+				{
+					count: 7,
+					height: 13,
+				},
+			],
+		};
+
+		beforeEach(async () => {
+			await usedHashOnionStore.set(context, address, defaultUsedHashOnionStoreObject);
+		});
+
+		it('Upsert provided usedHashInfo, corresponding to count and filters usedHashOnions having height greater than provided height', async () => {
```

Comment:
maybe add cases which tests below?
* adding to the last of the array
* adding to the middle of the array
* filter by finalizedHeight

Reply:
The test case actually covers, update and insert case, however, the filter by finalized heights is missing, I can have the filter by height test case covered in the same test case, but probably dividing them into three different test cases is a better idea.